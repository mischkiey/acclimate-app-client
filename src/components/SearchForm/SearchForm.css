.search-form-inputs-group {
    border: 1px dashed green;
    display: flex;
    flex-flow: column;
    margin-bottom: 10px;
}

.search-form-input-group {
    align-items: center;
    border: 1px dashed rgb(219, 213, 213);
    border-radius: 10px;
    display: flex;
    flex-flow: column;
    margin: 10px;
}

.search-input-image {
    height: 300px;
    width: 75%;
}

input {
    width: 50%;
}


@media all and (min-width: 800px) {
    .search-form-inputs-group {
        flex-flow: row wrap;
    }

    .search-input-image {
        height: 300px;
        width: 100%;
    }
};