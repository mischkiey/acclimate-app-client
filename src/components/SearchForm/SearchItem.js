import React from 'react';
import './SearchForm.css';

function SearchItem(props) {
    return (
        <div className=''>
            {/* <label htmlFor={props.disaster_name}>{props.disaster_name}</label>
            <input id={props.disaster_id} type='image' alt={`${props.disaster_name}`} src={`${props.disaster_image}`}/> */}
            <label htmlFor={props.disaster_id}>
                {props.disaster_name}
            </label>
            <input id={props.disaster_id} name='disaster' type='radio' value={props.disaster_id}/>
            <img alt={`${props.disaster_name}`} src={`${props.disaster_image}`} />
        </div>
    );
};

export default SearchItem;