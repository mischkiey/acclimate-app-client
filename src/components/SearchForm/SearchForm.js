import React, { Component } from 'react';
import SearchItem from './SearchItem';

class SearchForm extends Component {
    state = {
        disasters: ['Hurricane', 'Tsunami', 'Volcanic Eruption']
    }

    render () {
        const options = this.state.disasters.map((disaster, idx) => <SearchItem disaster={disaster} key={idx} />)

        return (
            <>
                <h2>Search Form</h2>
                <form>
                    {options}
                </form>
            </>
        )
    };
};

export default SearchForm;