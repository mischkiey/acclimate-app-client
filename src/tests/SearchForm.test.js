import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import SearchForm from '../components/SearchForm/SearchForm';

describe(`SearchForm Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>    
                    <SearchForm />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <SearchForm />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})