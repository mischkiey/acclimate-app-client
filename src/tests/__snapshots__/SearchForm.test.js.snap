// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Component Snapshot test Renders the UI as expected 1`] = `
Array [
  <h2
    className="center"
  >
    Select a Disaster Program
  </h2>,
  <form
    className=""
    onSubmit={[Function]}
  >
    
    <div
      className="search-form-inputs-group"
    />
    <button
      className="y-btn"
    >
      <i
        className="material-icons"
      >
        add_circle
      </i>
    </button>
  </form>,
]
`;
